### PR TITLE
add check for externalized processors

### PR DESCRIPTION
**Pull request by freben**
No changeset since it's covered by the previously merged externalizations

### CODE REVIEW

File: plugins/catalog-backend/src/service/CatalogBuilder.ts [hunk 1]
```diff
@@ -577,4 +579,80 @@ export class CatalogBuilder {
...
+        locationTypes.has(locationType) &&
+        !processorNames.has(processorName)
+      ) {
+        throw new Error(
+          `Your config contains a "catalog.locations" entry of type ${locationType}, \
+           but does not have the corresponding catalog processor ${processorName} installed. \
+           This processor used to be built into the catalog itself, but is now moved to an \
+           external module that has to be installed manually. Please follow the installation \
+           instructions at ${installationUrl} if you are using this ability, or remove the \
+           location from your app config if you do not.`,
```
Comment (Rugvip) [inline plugins/catalog-backend/src/service/CatalogBuilder.ts:607]: Perhaps we could also add an env var that one can set to ignore this check? On the off chance that someone has a custom processor to consumer these same locations. Something like `BACKSTAGE_CATALOG_SKIP_MISSING_PROCESSORS_CHECK`

Comment (freben) [reply plugins/catalog-backend/src/service/CatalogBuilder.ts:607]: sure i can add that

Comment (freben) [reply plugins/catalog-backend/src/service/CatalogBuilder.ts:611]: @Rugvip updated - also snuck in the api report warning-free-ness limitation while i was at it

Comment (Rugvip) [reply plugins/catalog-backend/src/service/CatalogBuilder.ts:611]: I'm thinking also mention the env var as an escape hatch in this message?

Comment (freben) [reply plugins/catalog-backend/src/service/CatalogBuilder.ts]: Ah of course, I'm rushing too much - added now

**Review by Rugvip — APPROVED** [review]
🎉

**Pull request merged**